src/app/clusters/webrtc-transport-provider-server/webrtc-transport-provider-server.cpp

```diff
@@ -0,0 +1,495 @@
+/*
+ *
+ *    Copyright (c) 2025 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#include "webrtc-transport-provider-server.h"
+
+#include <protocols/interaction_model/StatusCode.h>
+
+#include <app-common/zap-generated/attributes/Accessors.h>
+#include <app-common/zap-generated/cluster-enums.h>
+#include <app-common/zap-generated/cluster-objects.h>
+#include <app/AttributeAccessInterfaceRegistry.h>
+#include <app/CommandHandler.h>
+#include <app/CommandHandlerInterfaceRegistry.h>
+#include <app/EventLogging.h>
+#include <app/reporting/reporting.h>
+#include <platform/CHIPDeviceLayer.h>
+#include <platform/PlatformManager.h>
+
+#include <memory>
+
+using namespace chip;
+using namespace chip::app;
+
+using chip::Protocols::InteractionModel::Status;
+
+namespace {
+
+NodeId GetNodeIdFromCtx(const CommandHandler & commandHandler)
+{
+    auto descriptor = commandHandler.GetSubjectDescriptor();
+
+    if (descriptor.authMode != Access::AuthMode::kCase)
+    {
+        return kUndefinedNodeId;
+    }
+    return descriptor.subject;
+}
+
+} // anonymous namespace
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace WebRTCTransportProvider {
+
+WebRTCTransportProviderServer::WebRTCTransportProviderServer(Delegate * delegate, EndpointId endpointId) :
+    AttributeAccessInterface(MakeOptional(endpointId), WebRTCTransportProvider::Id),
+    CommandHandlerInterface(MakeOptional(endpointId), WebRTCTransportProvider::Id), mDelegate(delegate)
+{}
+
+WebRTCTransportProviderServer::~WebRTCTransportProviderServer()
+{
+    CommandHandlerInterfaceRegistry::Instance().UnregisterCommandHandler(this);
+    AttributeAccessInterfaceRegistry::Instance().Unregister(this);
+}
+
+CHIP_ERROR WebRTCTransportProviderServer::Init()
+{
+    // Register ourselves for attribute read/write
```

Comment:
These comments seem to re-state the code. Maybe remove/replace to something that explains extra? E.g. why does the CHI register return an error, but not the AAI one? maybe we should use the bool to return an error on false.